tests/test_phase_psf.py

```diff
@@ -541,72 +569,134 @@ def test_phase_gradient_shoot():
-        im_fft = psf.drawImage(nx=48, ny=48, scale=0.05)
+        im_fft = psf2.drawImage(nx=48, ny=48, scale=0.05)
 
-        shoot_moment = galsim.utilities.unweighted_moments(im_shoot)
-        fft_moment = galsim.utilities.unweighted_moments(im_fft)
+        # at this point, the atms should be different.
+        assert atm != atm2
 
-        for key in ['Mx', 'My']:
-            np.testing.assert_allclose(
-                    shoot_moment[key], fft_moment[key], rtol=0, atol=centroid_tolerance,
-                    err_msg='Phase gradient centroid {0} not close to fft centroid'.format(key))
+        shoot_moment = galsim.hsm.FindAdaptiveMom(im_shoot)
+        fft_moment = galsim.hsm.FindAdaptiveMom(im_fft)
 
-        for key in ['Mxx', 'Myy']:
-            np.testing.assert_allclose(
-                    shoot_moment[key]+offset, fft_moment[key], rtol=0, atol=second_moment_tolerance,
-                    err_msg='Phase gradient second moment {} not close to fft moment'.format(key))
+        # print()
+        # print()
+        # print()
+        # print(shoot_moment.observed_shape.g1)
+        # print(fft_moment.observed_shape.g1)
+
+        # import matplotlib.pyplot as plt
+        # fig, axes = plt.subplots(ncols=2)
+        # axes[0].imshow(im_shoot.array)
+        # axes[1].imshow(im_fft.array)
+        # plt.show()
 
         np.testing.assert_allclose(
-            shoot_moment['Mxy'], fft_moment['Mxy'], rtol=0, atol=second_moment_tolerance,
-            err_msg='Phase gradient second moment Mxy not close to fft moment')
+            shoot_moment.moments_centroid.x,
+            fft_moment.moments_centroid.x,
+            rtol=0, atol=centroid_tolerance,
+            err_msg='Phase gradient centroid x not close to fft centroid')
 
-        shoot_moments.append(shoot_moment)
-        fft_moments.append(fft_moment)
+        np.testing.assert_allclose(
+            shoot_moment.moments_centroid.y,
+            fft_moment.moments_centroid.y,
+            rtol=0, atol=centroid_tolerance,
+            err_msg='Phase gradient centroid y not close to fft centroid')
 
-    # Verify that shoot with rng=None runs
-    psf.shoot(100, rng=None)
+        np.testing.assert_allclose(
+            shoot_moment.moments_sigma,
+            fft_moment.moments_sigma*(1+size_bias),
+            rtol=0, atol=size_tolerance,
+            err_msg='Phase gradient sigma not close to fft sigma')
 
-    # Constraints on the ensemble should be tighter than for individual PSFs.
-    mean_shoot_moment = {}
-    mean_fft_moment = {}
-    for k in shoot_moments[0]:
-        mean_shoot_moment[k] = np.mean([sm[k] for sm in shoot_moments])
-        mean_fft_moment[k] = np.mean([fm[k] for fm in fft_moments])
+        np.testing.assert_allclose(
+            shoot_moment.moments_sigma,
+            vkMom.moments_sigma,
+            rtol=0.1, atol=0,
+            err_msg='Phase gradient sigma not close to infinite exposure analytic sigma'
+        )
```

Comment:
> I thought the infinite exposure sigma should be (a little) larger than the short-exposure-time realization.

I think that's probably true, but the main difference here between `__name__ == '__main__'` and `!= '__main__'` is actually the `screen_scale` difference of 0.1 vs 0.025.  (Sorry, I put that code way at the top of this function separate from the other differences).  I talked about this some at the DESC meeting, and it's the subject of one of the validation tests (not that you should have remembered, but in case it rings a bell) and I should have reiterated this here: the realized size of the FFT PSF depends fairly strongly on the scale of the phase screens; growing as you increase resolution.  

So I think we expect better and better agreement between FFT and both photon-shooting and VonKarman-convolved-with-Airy as screen_scale goes down.  Unfortunately, the running time for FFT PSF drawing scales pretty steeply like screen_scale^-2, so I made a fast version and a slow version, but where we expect a fairly significant size bias for the fast version.

Does that make sense?  (And I will add some commentary to the top of the test to reflect all this).